core/domain/activity_jobs_one_off.py

```diff
@@ -628,3 +632,128 @@ def reduce(key, values):
             yield (key, values)
         else:
             yield (key, len(values))
+
+
+class AddMissingCommitLogsJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that adds the missing commit log entry model for the corresponding
+    snapshot models.
+    """
+
+    # This list consists of the snapshot models whose associated commit log
+    # models are missing. The commit logs were found missing in a validation
+    # job.
+    SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS = [
+        exp_models.ExplorationRightsSnapshotMetadataModel,
+        question_models.QuestionSnapshotMetadataModel,
+        skill_models.SkillSnapshotMetadataModel
+    ]
+    MODEL_NAMES_TO_PROPERTIES = {
+        'QuestionSnapshotMetadataModel': {
+            'parent_model_class': question_models.QuestionModel,
+            'commit_log_model_class': (
+                question_models.QuestionCommitLogEntryModel),
+            'id_string_format': 'question-%s-%s',
+            'id_field': 'question_id',
+            'delete_method': question_services.delete_question
+        },
+        'SkillSnapshotMetadataModel': {
+            'parent_model_class': skill_models.SkillModel,
+            'commit_log_model_class': skill_models.SkillCommitLogEntryModel,
+            'id_string_format': 'skill-%s-%s',
+            'id_field': 'skill_id',
+            'delete_method': skill_services.delete_skill
+        },
+        'ExplorationRightsSnapshotMetadataModel': {
+            'parent_model_class': exp_models.ExplorationRightsModel,
+            'commit_log_model_class': exp_models.ExplorationCommitLogEntryModel,
+            'id_string_format': 'rights-%s-%s',
+            'id_field': 'exploration_id',
+            'delete_method': exp_services.delete_exploration
+        }
+    }
+    # "Public" is the default value for post_commit_status for the
+    # question and skill models.
+    MODEL_NAMES_WITH_DEFAULT_COMMIT_STATUS = [
+        'QuestionSnapshotMetadataModel', 'SkillSnapshotMetadataModel']
+    # For the exp rights commit log model, post_commit_status is assigned
+    # from the exp rights model.
+    MODEL_NAMES_WITH_COMMIT_STATUS_IN_RIGHTS = [
+        'ExplorationRightsSnapshotMetadataModel']
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return cls.SNAPSHOT_METADATA_MODELS_WITH_MISSING_COMMIT_LOGS
+
+    @staticmethod
+    def map(snapshot_model):
+        job_class = AddMissingCommitLogsJob
+        class_name = snapshot_model.__class__.__name__
```

Comment:
Suggest: ``model_class_name`` to be less ambiguous (otherwise: what class is this referring to?)

Reply:
Done.